src/ast/mod.rs

```diff
@@ -1359,6 +1409,137 @@ impl fmt::Display for Statement {
     }
 }
 
+/// Privileges granted in a GRANT statement or revoked in a REVOKE statement.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum Privileges {
+    /// All privileges applicable to the object type
+    All {
+        /// Optional keyword from the spec, ignored in practice
+        with_privileges_keyword: bool,
```

Comment:
Maybe worth removing it from the struct then?

Reply:
@alamb I erred on the side of losslessness here, but I wasn't sure to what degree that's a goal.

I guess there are already many cases where an implied default becomes explicit when parsed, and would then produce different output after a roundtrip if the input had omitted the default (eg. `cascade` in `Statement::Drop`).

If you think it would be cleaner I'm happy to PR removing this.

Reply:
If this field is needed to survive the round trip I think keeping it makes sense.